app/code/community/Iterable/TrackOrderPlaced/Helper/Data.php

```diff
@@ -1,106 +1,69 @@
-    const XML_PATH_TRANSACTIONAL_FORGOT_PASSWORD_CAMPAIGN_ID = 'transactional_email_options/forgot_password/campaign_id';
-    const XML_PATH_TRANSACTIONAL_NEWSLETTER_SUBSCRIBE_ENABLED = 'transactional_email_options/newsletter_subscribe/is_enabled';
-    const XML_PATH_TRANSACTIONAL_NEWSLETTER_SUBSCRIBE_CAMPAIGN_ID = 'transactional_email_options/newsletter_subscribe/campaign_id';
-    const XML_PATH_TRANSACTIONAL_NEWSLETTER_UNSUBSCRIBE_ENABLED = 'transactional_email_options/newsletter_unsubscribe/is_enabled';
-    const XML_PATH_TRANSACTIONAL_NEWSLETTER_UNSUBSCRIBE_CAMPAIGN_ID = 'transactional_email_options/newsletter_unsubscribe/campaign_id';
-
-    // disabled section
-    const XML_PATH_TRANSACTIONAL_ORDER_CONFIRM_DISABLED = 'transactional_email_options/order_confirm/is_disabled';
-
-    /** end transactional email xml config */
-
-    /** @var array Map of templateName -> (enabled, campaignId) */
-    private static $EMAIL_TEMPLATE_NAMES_TO_TRANSACTIONAL_EMAIL_OPTIONS = array(
-        'customer_create_account_email_template' => array(
-            self::XML_PATH_TRANSACTIONAL_ACCOUNT_CREATED_ENABLED,
-            self::XML_PATH_TRANSACTIONAL_ACCOUNT_CREATED_CAMPAIGN_ID
-        ),
-        'customer_password_forgot_email_template' => array(
-            self::XML_PATH_TRANSACTIONAL_FORGOT_PASSWORD_ENABLED,
-            self::XML_PATH_TRANSACTIONAL_FORGOT_PASSWORD_CAMPAIGN_ID
-        ),
-        'newsletter_subscription_success_email_template' => array(
-            self::XML_PATH_TRANSACTIONAL_NEWSLETTER_SUBSCRIBE_ENABLED,
-            self::XML_PATH_TRANSACTIONAL_NEWSLETTER_SUBSCRIBE_CAMPAIGN_ID
-        ),
-        'newsletter_subscription_un_email_template' => array(
-            self::XML_PATH_TRANSACTIONAL_NEWSLETTER_UNSUBSCRIBE_ENABLED,
-            self::XML_PATH_TRANSACTIONAL_NEWSLETTER_UNSUBSCRIBE_CAMPAIGN_ID
-        )
-    );
-
-    public static function getTransactionalEmailConfig() {
-        return self::$EMAIL_TEMPLATE_NAMES_TO_TRANSACTIONAL_EMAIL_OPTIONS;
-    }
-
-    private static $EMAIL_TEMPLATE_NAMES_TO_DISABLED_OPTIONS = array(
-        'sales_email_order_template' => self::XML_PATH_TRANSACTIONAL_ORDER_CONFIRM_DISABLED
-    );
-
-    public static function getDefaultEmailDisabledConfig() {
-        return self::$EMAIL_TEMPLATE_NAMES_TO_DISABLED_OPTIONS;
-    }
-
-    private function getDecodedMagentoApiToken() {
+    private function getDecodedMagentoApiToken()
+    {
         $magentoApiKey = Mage::getStoreConfig(self::XML_PATH_ITERABLE_API_KEY);
+
         return json_decode(base64_decode($magentoApiKey));
     }
 
-    private function getIterableApiToken() {
+    private function getIterableApiToken()
+    {
         $apiKeyJson = $this->getDecodedMagentoApiToken();
-        if ($apiKeyJson == NULL) {
-            return NULL;
+        if ($apiKeyJson == null) {
+            mage::throwException('API Key is not set');
```

Comment:
Valid point. I will adjust. Mote that i was going to withdraw rhis merge in favour of a new one with improved code and ability to push events as well.